tst/EnergyPlus/unit/WeatherManager.unit.cc

```diff
@@ -2214,3 +2214,384 @@ TEST_F(EnergyPlusFixture, epwHeaderTest)
     EXPECT_EQ(state->dataWeatherManager->DataPeriods(2).EnYear, 0);
     EXPECT_FALSE(state->dataWeatherManager->DataPeriods(2).HasYearData);
 }
+
+TEST_F(EnergyPlusFixture, DisplayWeatherMissingDataWarnings_TMYx)
+{
+
+    // Test for #9709
+
+    {
+        int WYear = 0;
+        int WMonth = 0;
+        int WDay = 0;
+        int WHour = 0;
+        int WMinute = 0;
+        Real64 DryBulb = 0.0;
+        Real64 DewPoint = 0.0;
+        Real64 RelHum = 0.0;
+        Real64 AtmPress = 0.0;
+        Real64 ETHoriz = 0.0;
+        Real64 ETDirect = 0.0;
+        Real64 IRHoriz = 0.0;
+        Real64 GLBHoriz = 0.0;
+        Real64 DirectRad = 0.0;
+        Real64 DiffuseRad = 0.0;
+        Real64 GLBHorizIllum = 0.0;
+        Real64 DirectNrmIllum = 0.0;
+        Real64 DiffuseHorizIllum = 0.0;
+        Real64 ZenLum = 0.0;
+        Real64 WindDir = 0.0;
+        Real64 WindSpeed = 0.0;
+        Real64 TotalSkyCover = 0.0;
+        Real64 OpaqueSkyCover = 0.0;
+        Real64 Visibility = 0.0;
+        Real64 CeilHeight = 0.0;
+        Real64 PrecipWater = 0.0;
+        Real64 AerosolOptDepth = 0.0;
+        Real64 SnowDepth = 0.0;
+        Real64 DaysSinceLastSnow = 0.0;
+        Real64 Albedo = 0.0;
+        Real64 LiquidPrecip = 0.0;
+        int PresWeathObs = 0;
+        Array1D_int PresWeathConds(9);
+        bool ErrorFound = false;
+
+        // Weather codes are present, taken from PRT_Faro.085540_IWEC.epw
+        std::string WeatherDataLine =
+            "2018,1,5,15,0,?9?9?9?9E0?9?9?9?9?9?9?9?9?9?9?9?9?9*9?9?9?9,16.60,13.50,82,100997,588,1415,369,195,14,189,22230,1274,22230,"
+            "10584,320,10.00,9,9,777.7,1050,0,909999999,37,0.0840,0,88,0.200,0.0,0.0";
```

Comment:
`0,909999999` => the 0 means the weather codes are present, which isn't the case for Chicago EPW for eg, but was the case for the defect file (and the PRT_Faro IWEC file too). Needed to get the error.